bindings/java/c/evmc-vm.c

```diff
@@ -15,23 +21,16 @@ JNIEXPORT jobject JNICALL Java_org_ethereum_evmc_EvmcVm_init(JNIEnv* jenv,
     jint rs = set_jvm(jenv);
     assert(rs == JNI_OK);
     // load the EVM
-    const char* filename = (*jenv)->GetStringUTFChars(jenv, jfilename, 0);
-    if (filename != NULL)
-    {
-        enum evmc_loader_error_code loader_error;
-        evm = evmc_load_and_create(filename, &loader_error);
-        if (evm == NULL || loader_error != EVMC_LOADER_SUCCESS)
-        {
-            const char* error_msg = evmc_last_error_msg();
-            jclass jclazz = (*jenv)->FindClass(jenv, "java/lang/AssertionError");
-            (*jenv)->ThrowNew(jenv, jclazz, error_msg ? error_msg : "Loading EVMC VM failed");
-        }
-        (*jenv)->ReleaseStringUTFChars(jenv, jfilename, filename);
-    }
-    else
+    const char* filename = (*jenv)->GetStringUTFChars(jenv, jfilename, NULL);
+    if (filename == NULL)
+        throw_java_assert(jenv, "JNI Error: filename cannot be NULL");
+    enum evmc_loader_error_code loader_error;
+    evm = evmc_load_and_create(filename, &loader_error);
+    (*jenv)->ReleaseStringUTFChars(jenv, jfilename, filename);
+    if (evm == NULL || loader_error != EVMC_LOADER_SUCCESS)
```

Comment:
```suggestion
    if (loader_error != EVMC_LOADER_SUCCESS)
```
Don't check `evm` for null, checking error code is enough.

Reply:
This was in the original version of the code.

Reply:
I can see that.

Reply:
Fixed.